usr/src/lib/libzfs/common/libzfs_pool.c

```diff
@@ -1851,22 +1852,39 @@ zpool_import_props(libzfs_handle_t *hdl, nvlist_t *config, const char *newname,
  * Scan the pool.
  */
 int
-zpool_scan(zpool_handle_t *zhp, pool_scan_func_t func)
+zpool_scan(zpool_handle_t *zhp, pool_scan_func_t func, pool_scrub_cmd_t cmd)
 {
 	zfs_cmd_t zc = { 0 };
 	char msg[1024];
+	int err;
 	libzfs_handle_t *hdl = zhp->zpool_hdl;
 
 	(void) strlcpy(zc.zc_name, zhp->zpool_name, sizeof (zc.zc_name));
 	zc.zc_cookie = func;
+	zc.zc_flags = cmd;
+
+	if (zfs_ioctl(hdl, ZFS_IOC_POOL_SCAN, &zc) == 0)
+		return (0);
+
+	err = errno;
 
-	if (zfs_ioctl(hdl, ZFS_IOC_POOL_SCAN, &zc) == 0 ||
-	    (errno == ENOENT && func != POOL_SCAN_NONE))
+	/* ECANCELED on a scrub means we resumed a paused scrub */
```

Comment:
Even though this comment makes everything clear with respect to the rest of the code, I found it a bit confusing that ECANCELED means resume a paused scrub. Skimming through other errnos the best alternative I could find is EADV, that I read as EADVANCE but apparently is EADVERTISE which is obscure.

That said, I wonder if we could make use of zc.zc_nvlist_dst in the kernel and back to the userland to deal with both the ECANCELED and the ENOENT case later in this code path, in a cleaner way. Did we consider that?